styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteInputsUtils.java

```diff
@@ -24,30 +24,111 @@
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toSet;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import flyteidl.core.Interface;
 import flyteidl.core.Literals;
 import flyteidl.core.Types;
-
+import flyteidl.core.Types.LiteralType;
+import flyteidl.core.Types.LiteralType.TypeCase;
+import flyteidl.core.Types.SimpleType;
+import java.time.Duration;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
-
 public class FlyteInputsUtils {
 
+  private static final Set<TypeCase> COMPLEX_TYPES =
+      Set.of(TypeCase.COLLECTION_TYPE, TypeCase.MAP_VALUE_TYPE);
+
+  private static final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
```

Comment:
We have https://github.com/spotify/styx/blob/master/styx-common/src/main/java/com/spotify/styx/serialization/Json.java#L66 already.